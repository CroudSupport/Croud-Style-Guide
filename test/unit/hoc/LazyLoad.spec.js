import Vue from 'vue/dist/vue.common'
import Lazy from '../../../src/components/shared/hoc/LazyLoad'

window.IntersectionObserver = jest.fn(() => ({
    observe: jest.fn(),
}))

const Constructor = Vue.extend(Lazy)
const mounted = jest.fn()
const renderer = Vue.compile('<div>Test</div>')
const patsy = {
    render: renderer.render,
    staticRenderFns: renderer.staticRenderFns,
    mounted,
}

const vm = new Constructor()

vm.$slots.default = [vm.$createElement(patsy)]
vm.$mount()

describe('Lazy Load', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })

    it('should lazy load when in view', (done) => {
        vm.outOfView = false
        vm.$nextTick(() => {
            expect(vm.$el).toMatchSnapshot()
            expect(mounted).toHaveBeenCalled()
            done()
        })
    })
})
