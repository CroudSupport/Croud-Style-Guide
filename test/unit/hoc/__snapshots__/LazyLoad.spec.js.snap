// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lazy Load should lazy load when in view 1`] = `
<div>
  <div>
    Test
  </div>
</div>
`;

exports[`Lazy Load should match the snapshot 1`] = `
<div
  class="placeholder"
>
  
</div>
`;
